pkg/controller/provider/container/ocp/collection.go

```diff
@@ -334,7 +350,12 @@ func (r *VM) Reconcile(ctx context.Context) (err error) {
 		err = liberr.Wrap(err)
 		return
 	}
-	defer tx.End()
+	defer func() {
+		err = tx.End()
+		if err != nil {
+			err = liberr.Wrap(err)
+		}
+	}()
```

Comment:
wouldn't it make sense to extract this to a separate function?

Reply:
yeah, that's what I meant (even though I'm not sure why you placed the function in a different place rather than having it here as a "private" function)